src/Store/VimStoreConnector.re

```diff
@@ -138,18 +138,8 @@ let start =
     );
 
   let _: unit => unit =
-    Vim.onMessage((priority, title, msg) => {
-      open Vim.Types;
-      let (priorityString, kind) =
-        switch (priority) {
-        | Error => ("ERROR", Notification.Error)
-        | Warning => ("WARNING", Notification.Warning)
-        | Info => ("INFO", Notification.Info)
-        };
-
-      Log.debugf(m => m("Message - %s [%s]: %s", priorityString, title, msg));
-
-      dispatch(ShowNotification(Notification.create(~kind, msg)));
+    Vim.onMessage((priority, title, message) => {
+      dispatch(VimMessageReceived({priority, title, message}))
```

Comment:
Here too, instead of making decisions in callbacks scattered all over the place, dispatch an event-like message and do the decision-making and side-effects in in the store/update function.